src/App.js

```diff
@@ -0,0 +1,234 @@
+            $checkbox.setAttribute('checked', false);
+            this.todoComplete[index] = false;
+        }
+        this.updateLocalStorage();
+    }
+    updateLocalStorage = () => {
+        localStorage.setItem('todoList', JSON.stringify(this.todoList));
+        localStorage.setItem('todoComplete', JSON.stringify(this.todoComplete));
+        this.todoList = JSON.parse(localStorage.getItem('todoList'));
+        this.todoComplete = JSON.parse(localStorage.getItem('todoComplete'));
+    }
+
+    renderTodo = todo => {
+        const $todoItem = document.createElement('li');
+        $todoItem.innerHTML = `
+            <div class="view">
+                <input class="toggle" type="checkbox"/>
+                <label class="label">${todo}</label>
+                <button class="destroy"></button>
+            </div>
+            <input class="edit" value="${todo}" />
+            `;
+
+        this.$todoList.appendChild($todoItem);
+        return $todoItem;
+    }
+
+    render = () => {
+        this.$todoList.innerHTML = ``;
+        let cnt = 0;
+
+        if (this.todoFilter === 'all') {
+            this.todoList.forEach(($item, index) => {
+                const $todoItem = this.renderTodo($item);
+                if (this.todoComplete[index]) {
+                    $todoItem.classList.add('completed');
+                    $todoItem.querySelector('input.toggle').setAttribute('checked', true);
+                }
+                cnt++;
+            })
+        }
+        else if (this.todoFilter === 'active') {
+            this.todoList.forEach(($item, index) => {
+                if (!this.todoComplete[index]) {
+                    this.renderTodo($item);
+                    cnt++;
+                }
+            })
+        }
+        else if (this.todoFilter === 'completed') {
+            this.todoList.forEach(($item, index) => {
+                if (this.todoComplete[index]) {
+                    const $todoItem = this.renderTodo($item);
+                    $todoItem.classList.add('completed');
+                    $todoItem.querySelector('input.toggle').setAttribute('checked', true);
+                    cnt++;
+                }
+            })
+        }
+        this.updateTodoCount(cnt);
+    }
+
+    convertToEditMode = $todoItem => {
+        const editing = $todoItem.classList.contains('editing');
+
+        if (!editing) {
+            $todoItem.classList.add('editing');
+        }
+    }
+    cancelEditMode = $todoItem => {
+        const editing = $todoItem.classList.contains('editing');
+
+        if (editing) {
+            $todoItem.classList.remove('editing');
+        }
+    }
+    editTodo = ($item, index) => {
+        $item.classList.remove('editing');
+        this.todoList[index] = $item.querySelector('input.edit').value;
+        $item.querySelector('label.label').innerText = this.todoList[index];
+        $item.querySelector('input.edit').value = '';
```

Comment:
```edit```의 밸류가 공백으로 변경되서 아이템을 한번 수정한 후 재수정을 위해 다시 더블클릭 했을 때 해당 아이템의 내용이 공백으로 보이는 버그가 발생하는거 같습니다. 아래 코드처럼 ```edit```에도  변경된 값을 담아주는 과정이 필요할거 같습니다!
```suggestion
        $item.querySelector('input.edit').value = $item.querySelector('input.edit').value;
```

Reply:
아 그런 문제가 있었군요. 수정하였습니다. 감사합니다!